libraries/src/Console/CreatePublicFolderCommand.php

```diff
@@ -0,0 +1,301 @@
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * Method to get a value from option
+     *
+     * @param   string  $option    set the option name
+     * @param   string  $question  set the question if user enters no value to option
+     * @param   bool    $required  is it required
+     *
+     * @return  string
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function getStringFromOption($option, $question, $required = true): string
+    {
+        $answer = (string) $this->cliInput->getOption($option);
+
+        while (!$answer && $required) {
+            $answer = (string) $this->ioStyle->ask($question);
+        }
+
+        if (!$required) {
+            $answer = (string) $this->ioStyle->ask($question);
+        }
+
+        return $answer;
+    }
+
+    /**
+     * Configure the IO.
+     *
+     * @param   InputInterface   $input   The input to inject into the command.
+     * @param   OutputInterface  $output  The output to inject into the command.
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    private function configureIO(InputInterface $input, OutputInterface $output)
+    {
+        $this->cliInput = $input;
+        $this->ioStyle  = new SymfonyStyle($input, $output);
+    }
+
+    /**
+     * Configure the command.
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    protected function configure(): void
+    {
+        $help = "<info>%command.name%</info> will create a public folder
+		\nUsage: <info>php %command.full_name%</info>";
+
+        $this->addOption('public-folder', null, InputOption::VALUE_REQUIRED, 'public folder absolute path');
+        $this->setDescription('Create a public folder');
+        $this->setHelp($help);
+    }
+
+    private function createSymlink($source, $dest)
+    {
+        if (!symlink($source, $dest)) {
+            throw new \Exception('Unable to symlink the file: ' . str_replace(JPATH_ROOT, '', $source), 200);
+        }
+    }
+
+    private function createFile($path, $content)
+    {
+        if (!file_put_contents($path, $content)) {
+            throw new \Exception('Unable to create the file: ' . str_replace(JPATH_ROOT, '', $path), 200);
+        }
+    }
+
+    private function createPublicFolder($folder): bool
+    {
+        if (!is_dir($folder) && !mkdir($folder, 0755, true)) {
+            throw new \Exception('The given directory doesn\'t exist or not accessible due to wrong permissions', 200);
+        }
+
+        // Create the required folders
+        if (
+            !mkdir($folder . '/administrator/components/com_joomlaupdate', 0755, true)
+            || !mkdir($folder . '/administrator/includes', 0755, true)
+            || !mkdir($folder . '/api/includes', 0755, true)
+            || !mkdir($folder . '/includes', 0755)
+        ) {
+            throw new \Exception('Unable to write on the given directory, check the permissions', 200);
```

Comment:
```suggestion
            throw new \Exception('Unable to write on the given directory, check the permissions', 200);
```

should that be "unable to create the ..."

you call it a folder elsewhere not a directory

Reply:
Legit, tbh let's wait for someone from the security/maintainers validate this before wasting our time here